api/clients/v2/payload_disperser.go

```diff
@@ -0,0 +1,222 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/api/clients/codecs"
+	"github.com/Layr-Labs/eigenda/api/clients/v2/verification"
+	dispgrpc "github.com/Layr-Labs/eigenda/api/grpc/disperser/v2"
+	"github.com/Layr-Labs/eigenda/common/geth"
+	core "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+// PayloadDisperser provides the ability to disperse payloads to EigenDA
+//
+// This struct is goroutine safe.
+type PayloadDisperser struct {
+	logger          logging.Logger
+	config          *PayloadDisperserConfig
+	codec           codecs.BlobCodec
+	disperserClient DisperserClient
+	certVerifier    verification.ICertVerifier
+}
+
+// BuildPayloadDisperser builds a PayloadDisperser from config structs
+func BuildPayloadDisperser(
+	logger logging.Logger,
+	payloadDisperserConfig *PayloadDisperserConfig,
+	disperserClientConfig *DisperserClientConfig,
+	// signer to sign blob dispersal requests
+	signer core.BlobRequestSigner,
+	// prover is used to compute commitments to a new blob during the dispersal process
+	//
+	// IMPORTANT: it is permissible for the prover parameter to be nil, but operating with this configuration
+	// puts a trust assumption on the disperser. With a nil prover, the disperser is responsible for computing
+	// the commitments to a blob, and the PayloadDisperser doesn't have a mechanism to verify these commitments.
+	//
+	// TODO: In the future, an optimized method of commitment verification using fiat shamir transformation will
+	//  be implemented. This feature will allow a PayloadDisperser to offload commitment generation onto the
+	//  disperser, but the disperser's commitments will be verifiable without needing a full-fledged prover
+	prover encoding.Prover,
+	accountant *Accountant,
+	ethConfig geth.EthClientConfig,
+) (*PayloadDisperser, error) {
+
+	codec, err := codecs.CreateCodec(
+		payloadDisperserConfig.PayloadPolynomialForm,
+		payloadDisperserConfig.BlobEncodingVersion)
+	if err != nil {
+		return nil, fmt.Errorf("create codec: %w", err)
+	}
+
+	disperserClient, err := NewDisperserClient(disperserClientConfig, signer, prover, accountant)
+	if err != nil {
+		return nil, fmt.Errorf("new disperser client: %s", err)
+	}
+
+	ethClient, err := geth.NewClient(ethConfig, gethcommon.Address{}, 0, logger)
```

Comment:
what benefits does using our wrapper client provide vs e.g just using standard `go-ethereum/ethclient`? at first glance this client seems well optimized for write capabilities which we'd never do when verifying. Would there be any value in leveraging the `InstrumentedEthClient` since it provides telemetry wrappers which may be insightful to a customer running proxy? Or perhaps for better future proofing we can leverage the `MultiHomingClient` since it provides redundancy support with multiple RPC providers?

Reply:
The wrapper client is need to pass into `NewContractEigenDABlobVerifierCaller`, in the constructor of `CertVerifier` https://github.com/litt3/eigenda/blob/70c0b92985bf53d5b8cef3542c2ee945d598f96e/api/clients/v2/verification/cert_verifier.go#L49

Regarding which flavor of eth client to use: I'm leaning toward just accepting an `EthClient` as a parameter, instead of constructing one here. That way the caller can just choose whichever type is best suited. Thoughts?